camel-quarkus-git: integration test fail if gpg signing is enabled

In my set-up I have GPG signing enabled on git, with the following config in .gitconfig

```
[commit]
    gpgsign = true
[gpg]
    program = gpg2
```

The git integration test fails with the following exception:

```
Caused by: org.eclipse.jgit.api.errors.ServiceUnavailableException: Signing service is not available
	at org.eclipse.jgit.api.CommitCommand.call(CommitCommand.java:247)
	at org.apache.camel.component.git.producer.GitProducer.doCommit(GitProducer.java:336)
	at org.apache.camel.component.git.producer.GitProducer.process(GitProducer.java:121)
	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:66)
	at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:218)
	at org.apache.camel.impl.engine.SharedCamelInternalProcessor$1.process(SharedCamelInternalProcessor.java:112)
	at org.apache.camel.impl.engine.DefaultAsyncProcessorAwaitManager.process(DefaultAsyncProcessorAwaitManager.java:83)
	at org.apache.camel.impl.engine.SharedCamelInternalProcessor.process(SharedCamelInternalProcessor.java:109)
	at org.apache.camel.support.cache.DefaultProducerCache.send(DefaultProducerCache.java:189)
	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:176)
	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:172)
	at org.apache.camel.impl.engine.DefaultProducerTemplate.send(DefaultProducerTemplate.java:158)
	at org.apache.camel.impl.engine.DefaultProducerTemplate.sendBodyAndHeader(DefaultProducerTemplate.java:229)
```

--- a/integration-tests/git/src/main/java/org/apache/camel/quarkus/component/git/it/GitResource.java
+++ b/integration-tests/git/src/main/java/org/apache/camel/quarkus/component/git/it/GitResource.java
@@ -16,6 +16,7 @@
  */
 package org.apache.camel.quarkus.component.git.it;
 
+import java.io.IOException;
 import java.net.URI;
 import java.nio.file.Files;
 import java.nio.file.Paths;
@@ -23,6 +24,7 @@ import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
 import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
 import javax.inject.Inject;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
@@ -33,10 +35,12 @@ import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
-import org.apache.camel.ConsumerTemplate;
+import io.quarkus.runtime.StartupEvent;
 import org.apache.camel.ProducerTemplate;
 import org.apache.camel.component.git.GitConstants;
+import org.eclipse.jgit.errors.ConfigInvalidException;
 import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.util.SystemReader;
 
 @Path("/git")
 @ApplicationScoped
@@ -45,8 +49,10 @@ public class GitResource {
     @Inject
     ProducerTemplate producerTemplate;
 
-    @Inject
-    ConsumerTemplate consumerTemplate;
+    public void init(@Observes StartupEvent startupEvent) throws IOException, ConfigInvalidException {
+        // Avoid consuming any local git config that may affect the tests
+        SystemReader.getInstance().getUserConfig().clear();
+    }
 
     @Path("/init/{repoName}")
     @POST

